File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    mode = 0
    ret = ""
    for i, char in enumerate(code):
        if mode == 0:
            if char != "1" and i%2 == 0: 
                ret += char
            elif char == "1":
                mode = 1
        else:
            if char != "1" and i%2 == 1:
                ret += char
            elif char == "1":
                mode = 0 
                
    return "EMPTY" if ret == "" else ret

File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        String answer = "";
        String code_peace = "";
        int mode = 0;
        
        for(int i = 0; i < code.length(); i++) {
            code_peace = String.valueOf(code.charAt(i));
            if(mode == 0) {
                if(!code_peace.equals("1") && i%2 == 0) {
                    answer += code_peace;
                } else if(code_peace.equals("1")) {
                    mode = 1;
                }
            } else if(mode == 1) {
                if(!code_peace.equals("1") && i%2 != 0) {
                    answer += code_peace;
                } else if(code_peace.equals("1")) {
                    mode = 0;
                }
            }
        }  
        return answer == "" ? "EMPTY" : answer;
    }
}